tldr/mcp_server.py

```
@@ -64,17 +89,44 @@ def _ensure_daemon(project: str, timeout: float = 10.0) -> None:
+                # LK_NBLCK is non-blocking; we loop with a 10s timeout
+                lock_start = time.time()
+                lock_timeout = 10.0  # Same as startup.py
+                while True:
+                    try:
+                        msvcrt.locking(lock_file.fileno(), msvcrt.LK_NBLCK, 10)
+                        break
+                    except OSError:
+                        if time.time() - lock_start > lock_timeout:
+                            raise RuntimeError(
+                                f"Timeout acquiring lock on {lock_path} after {lock_timeout}s"
+                            )
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)